controllers/module_reconciler.go

```diff
@@ -280,6 +324,10 @@ func (r *ModuleReconciler) handleBuild(ctx context.Context,
 
 	logger.Info("Image not pull-able; building in-cluster")
 
+	if err := r.injectDTKToBuildIfNeeded(ctx, kernelVersion, km.Build); err != nil {
```

Comment:
This is a little problematic: km.Build can be nil.

Reply:
Will add a validation to skip `nil`.
Thanks.

Reply:
you need more then checking for nil: you need to decide what Build to use, the one in the mapping or the global one. there is a function that does just that: GetRelevantBuild, and it is called at the beginning of Sync function. This is one of the reasons that i asked you to consider whether the inject function needs to be executed in the Sync, since it is part of creating the Build

Reply:
Done.